src/openrct2/paint/tile_element/Paint.LargeScenery.cpp

```diff
@@ -233,6 +233,9 @@ void large_scenery_paint(paint_session* session, uint8_t direction, uint16_t hei
 
     uint32_t image_id = (sequenceNum << 2) + entry->image + 4 + direction;
     rct_large_scenery_tile* tile = &entry->large_scenery.tiles[sequenceNum];
+    if (tile == nullptr)
```

Comment:
Is this actually possible? Isn't the address always going to real. Perhaps we should be checking instead that sequence number is within tiles size

Reply:
If it isn't possible, how do you explain #14217 crashing on line 259/262, where it references `tile`? ;-)

Reply:
I just don't think it would create an npe it should be creating some bad memory access instead.

Reply:
@Gymnasiast make sure to check the `fault.address` which on this issue is 0x07, so not a null pointer exception but still an access violation. This fix will not resolve the issue, unless the out of bounds memory just happened to be 0. What I suggest doing is creating a helper method on LargeSceneryElement that returns `nullptr` if sequenceNum is out of bounds.

Reply:
It's hard to tell, since I cannot reproduce the crash myself. Can't we just check for both?

Reply:
OK, should I check using `std::size(tiles)`?

Reply:
`rct_large_scenery_entry` doesn't have the number of tiles, but `LargeSceneryObject` does. You could either fetch that instance to get the total number of tiles, or you could add a new field to `rct_large_scenery_entry`. Although technically we should be moving towards the new Object classes for information rather than continue to use the old entry structs.

Reply:
OK, that should be it.